integration-test/src/test/java/org/sagebionetworks/ITStorageLocation.java

```diff
@@ -0,0 +1,152 @@
+import org.sagebionetworks.client.SynapseClient;
+import org.sagebionetworks.client.SynapseClientImpl;
+import org.sagebionetworks.client.exceptions.SynapseBadRequestException;
+import org.sagebionetworks.client.exceptions.SynapseException;
+import org.sagebionetworks.repo.model.Project;
+import org.sagebionetworks.repo.model.file.UploadDestinationLocation;
+import org.sagebionetworks.repo.model.project.ExternalObjectStorageLocationSetting;
+import org.sagebionetworks.repo.model.project.ExternalS3StorageLocationSetting;
+import org.sagebionetworks.repo.model.project.ExternalStorageLocationSetting;
+import org.sagebionetworks.repo.model.project.ProjectSettingsType;
+import org.sagebionetworks.repo.model.project.ProxyStorageLocationSettings;
+import org.sagebionetworks.repo.model.project.S3StorageLocationSetting;
+import org.sagebionetworks.repo.model.project.UploadDestinationListSetting;
+import org.sagebionetworks.util.ContentDispositionUtils;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+
+import static org.junit.Assert.assertEquals;
+
+// PFLM-5985 - Creating storage locations with UploadType=null, associating them via project setting to a project, and
+// then calling getUploadDestinationLocations will result in a NullPointerException. This is now fixed, and this test
+// verifies the fix.
+public class ITStorageLocation {
+	private static Long userToDelete;
+	private static StackConfiguration config;
+	private static SynapseAdminClient adminSynapse;
+	private static SynapseClient synapse;
+	private static SynapseS3Client synapseS3Client;
+
+	@BeforeClass
+	public static void beforeClass() throws Exception {
+		config = StackConfigurationSingleton.singleton();
+		// Create a user
+		adminSynapse = new SynapseAdminClientImpl();
+		SynapseClientHelper.setEndpoints(adminSynapse);
+		adminSynapse.setUsername(StackConfigurationSingleton.singleton().getMigrationAdminUsername());
+		adminSynapse.setApiKey(StackConfigurationSingleton.singleton().getMigrationAdminAPIKey());
+		adminSynapse.clearAllLocks();
+
+		synapse = new SynapseClientImpl();
+		userToDelete = SynapseClientHelper.createUser(adminSynapse, synapse);
+
+		synapseS3Client = AwsClientFactory.createAmazonS3Client();
+	}
+
+	@Before
+	public void before() throws SynapseException {
+		adminSynapse.clearAllLocks();
+	}
+
+	@AfterClass
+	public static void afterClass() {
+		try {
+			adminSynapse.deleteUser(userToDelete);
+		} catch (SynapseException e) {
+			// Ignore possible exceptions
+		}
+	}
+
+	@Test(expected = SynapseBadRequestException.class)
+	public void externalObjectStorageMustHaveUploadType() throws SynapseException {
+		ExternalObjectStorageLocationSetting storageLocationSetting = new ExternalObjectStorageLocationSetting();
+		storageLocationSetting.setBucket("some-bucket");
+		storageLocationSetting.setEndpointUrl("https://someurl.com");
+		storageLocationSetting.setUploadType(null);
```

Comment:
Follow-up: Is there an obvious UploadType for these particular storage location types?